### PR TITLE
Common aggregation

### PR DESCRIPTION
**Pull request by espegun**
Aggregation module for review.

### CODE REVIEW

File: common/aggregate_dfs.py [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+    _check_data_point_validity(df, data_points)
+
+    df_ratios = df.copy()
+    sums = df[data_points].sum(axis=1)
+
+    for dp in data_points:
+        col_name = '{dp}_ratio'.format(dp=dp)
+        df_ratios[col_name] = df_ratios[dp] / sums
+
+    return df_ratios
```
Comment (tordsk) [inline common/aggregate_dfs.py]: Note that dataset have different definitions on how ratio should be calculated. E.g. the ratio in `innvandrere_status` is calculated by `value / total population`. And the row does not sum up to total population

Comment (tordsk) [reply common/aggregate_dfs.py]: Maybe another function is needed in that case

File: common/aggregate_dfs.py [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+    This function checks if the data set is complete with entries for all years for all sub_districts.
+
+    Args:
+        df (pd.DataFrame): The input DataFrame.
+
+    Returns:
+        None
+    """
+
+    COLS = ['sub_district', 'date']
```
Comment (oyvinyg) [inline common/aggregate_dfs.py]: We decided to use 'delbydelid' as column name for subdistrict

File: common/aggregate_dfs.py [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+                raise ValueError('Column {col} is not found in the DataFrame header'.format(col=col))
+    # Input check - how
+    if how not in ['inner', 'left', 'right', 'outer']:
+        raise ValueError('The argument how needs to be either "inner", "left", "right" or "outer".')
+    # Input check - suffixes
+    if len(suffixes) != 2:
+        print('Suffixes={s}'.format(s=str(suffixes)))
+        raise ValueError('The length of suffixes is {length}, not 2 as expected.'.format(length=len(suffixes)))
+
+    df_merged = pd.merge(df1, df2, how=how, on=['date', 'district', 'sub_district'], suffixes=['_1', '_2'])
```
Comment (oyvinyg) [inline common/aggregate_dfs.py]: Here it should either be `suffixes=suffixes`, or the suffixes argument above should be removed.

File: common/aggregate_dfs.py [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+    It is a requirement that both DataFrames contain the columns 'date', 'district' and 'sub_district'.
+
+    Args:
+        df1 (pd.DataFrame): A DataFrame to be merged.
+        df2 (pd.DataFrame): A DataFrame to be merged.
+        how (str): Join method. Alternatives: 'inner', 'left', 'right', 'outer'.
+        suffixes (list of str): Suffixes to be added to new column if there are overlapping column names in df1 and df2.
+
+    Returns:
+        df_merged (pd.DataFrame): The original DataFrame with additional columns for ratios.
```
Comment (oyvinyg) [inline common/aggregate_dfs.py]: Maybe change the description of the return value. As far as I can see this method does not apply ratio columns.

File: tests/test_common_aggregate_dfs.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+        aggregation = {'agg_func': 'wmean',
+                       'data_points': 'mean_income',
+                       'data_weights': 'inhabitants'}
+        groupby = ['date']
+        df_act = aggregate_dfs._wmean(df_org, aggregation, groupby)
+        df_act, df_exp = _sort_dfs([df_act, df_agg_Oslo.copy()])
+
+        for col in df_act.columns:
+            self.assertListEqual(list(df_act[col]), list(df_exp[col]))
```
Comment (oyvinyg) [inline tests/test_common_aggregate_dfs.py]: I think that if you use `self.assertCountEqual(list(df_act[col]), list(df_exp[col]))` here you will not need to sort the rows of the dataframe

**Pull request merged**